lib/traject/config/sirsi_config.rb

```diff
@@ -1282,9 +1283,146 @@ def extract_isbn(value)
+      next if sub8.empty?
+      link_num, seq_num = sub8.split('.').map(&:to_i)
+
+      if mhld_field.most_recent863link_num < link_num || (
+        mhld_field.most_recent863link_num == link_num && mhld_field.most_recent863seq_num < seq_num
+      )
+        mhld_field.most_recent863link_num = link_num
+        mhld_field.most_recent863seq_num = seq_num
+        mhld_field.most_recent863 = field
+      end
+    when '866'
+      mhld_field.fields866 << field
+    when '867'
+      mhld_field.fields867 << field
+    when '868'
+      mhld_field.fields868 << field
+    end
+  end
+  accumulator.concat mhld_results.concat add_values_to_result(mhld_field)
+end
+
+def add_values_to_result(mhld_field)
+  return [] if mhld_field.library.nil?
+  latest_recd_out = false
+  has866 = false
+  has867 = false
+  has868 = false
+  mhld_results = []
+  mhld_field.fields866.each do |f|
+    sub_a = f.subfields.select do |sf|
+      %w[a].include? sf.code
+    end.collect(&:value).join('')
+
+    mhld_field.library_has = library_has(f)
+    if sub_a.end_with?('-')
+      unless latest_recd_out
+        latest_received = mhld_field.latest_received
+        latest_recd_out = true
+      end
+    end
+    has866 = true
+    mhld_results << mhld_field.display(latest_received)
+  end
+  mhld_field.fields867.each do |f|
+    mhld_field.library_has = "Supplement: #{library_has(f)}"
+    latest_received = mhld_field.latest_received unless has866
+    has867 = true
+    mhld_results << mhld_field.display(latest_received)
+  end
+  mhld_field.fields868.each do |f|
+    mhld_field.library_has = "Index: #{library_has(f)}"
+    latest_received = mhld_field.latest_received unless has866
+    has868 = true
+    mhld_results << mhld_field.display(latest_received)
+  end
+  if !has866 && !has867 && !has868
+    if mhld_field.df852has_equals_sf
+      latest_received = mhld_field.latest_received
+    end
+    mhld_results << mhld_field.display(latest_received)
+  end
+
+  mhld_results
+end
+
+def library_has(field)
+  field.subfields.select do |sf|
+    %w[a z].include? sf.code
+  end.collect(&:value).join(' ')
+end
+
+to_field 'bookplates_display' do |record, accumulator|
+  Traject::MarcExtractor.new('979').collect_matching_lines(record) do |field, spec, extractor|
+    file = field.subfields.select { |sf| sf.code == 'c' }.collect(&:value).first
+    next if file =~ /no content metadata/i
+    fund_name = field.subfields.select { |sf| sf.code == 'f' }.collect(&:value).first
+    druid = field.subfields.select { |sf| sf.code == 'b' }.collect(&:value).first.split(':')
+    text = field.subfields.select { |sf| sf.code == 'd' }.collect(&:value).first
+    accumulator << [fund_name, druid[1], file, text].join(' -|- ')
+  end
+end
+to_field 'fund_facet' do |record, accumulator|
+  Traject::MarcExtractor.new('979').collect_matching_lines(record) do |field, spec, extractor|
+    file = field.subfields.select { |sf| sf.code == 'c' }.collect(&:value).first
```

Comment:
`field.subfields.select { |sf| sf.code == 'c' }.collect(&:value).first` may be equivalent to `field['c']`?

Reply:
Thanks for the catch..changes made and pushed.